Formula/golangci-lint.rb

```diff
@@ -0,0 +1,39 @@
+class GolangciLint < Formula
+  desc "Fast linters runner for Go"
+  homepage "https://golangci.com"
+  url "https://github.com/golangci/golangci-lint.git",
+    :tag      => "v1.29.0",
+    :revision => "6a689074bf17fd4ae1db779a74dba821a162b6c8"
+  license "GPL-3.0"
+
+  depends_on "go" => :build
+
+  def install
+    commit = Utils.safe_popen_read("git", "rev-parse", "--short=7", "HEAD").chomp
+    ldflags = %W[
+      -s -w
+      -X main.version=#{version}
+      -X main.commit=#{commit}
+      -X main.date=#{Time.now.utc.rfc3339}
+    ].join(" ")
+
+    system "go", "build", *std_go_args, "-ldflags", ldflags, "./cmd/golangci-lint"
+
+    output = Utils.safe_popen_read({ "SHELL" => "bash" }, "#{bin}/golangci-lint", "completion", "bash")
+    (bash_completion/"golangci-lint").write output
+
+    output = Utils.safe_popen_read({ "SHELL" => "zsh" }, "#{bin}/golangci-lint", "completion", "zsh")
+    (zsh_completion/"golangci-lint").write output
+  end
+
+  test do
+    str_version = shell_output("#{bin}/golangci-lint --version")
+    str_help = shell_output("#{bin}/golangci-lint --help")
+    str_default = shell_output("#{bin}/golangci-lint")
+
+    assert_match "golangci-lint has version #{version} built from", str_version
+    assert_match "Usage:", str_help
+    assert_match "Available Commands:", str_help
+    assert_match str_default, str_help
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
I left version and help checks (they are working and checking if software works), in addition to them I have added functionality check for metal inter.